free-2.md

```diff
@@ -11,41 +11,88 @@ _What need is met by your idea? Who are you helping? What is that person's
 _Describe the project's domain in five words._
 
+Software development program compilation/linking
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
 
+The user might interact with the language by specifying facts about the different
+C/C++ files they want to compile/link and then which compiler/tags they want to use.
+I am imagining a sort of text file where these facts are entered and then the language
+understands it and makes the correct calls. I think this is the right way to interact
+with the language since it is similar to how make files currently work but removes
+some of the more complicated aspects like having to understand exactly how to crate
+the CLI calls in the make file. 
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+When a program runs, the different facts written down will be accumulated to determine 
+the structure of the code and then the proper compiler commands and tags will be used to
+compile/link everything. I think many errors could occur with not linking proper statements
+or not having all the edependiencies labeled and they would probably be communicated to 
+the user during the compilation process since it may be difficult to know if a file is missing
+or not. 
 
 ### Expressiveness
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
 
+Compiling/linking programs should be extremely easy. One should be able to 
+control which compiler they use and with what Command line arguments it is 
+run with. Also it should be easy to tell if a mistake has been made with the
+file linking process but that may be tricky to implement. Honestly it should be
+impossible/difficult to do anything else. Maybe it should just be difficult
+to add a looping functionality so as to allow for many files with similar names
+to be looped through?
 
 ### Related work
 _Are there any other DSLs in this domain? If not, describe how you know there
 aren't and conjecture why not. If so, describe them and provide links. How well 
 do they address the need? Are there any particularly admirable qualities of the
 language? Are there parts of the language you think could be improved?_
 
+I Google'd quite a bit to try and find any other DSL in this domain (other than Make)
+and I couldn't find anything. This kind of suprises me but I guess enough people have
+dealt with Make so far that it has not been worth creating another DSL for it.
+Now I personally do not believe that there is nothing like Make but I could not
+find it on google and that is the main way I know of discovering DSLs. There at least
+doesnt seem to be any popular DSL in this domain (other than make) so I assume this is
```

Comment:
See CMake and SCons as above. Also the autoconf/automake toolchain. There are more, but those are probably the most common.